### PR TITLE
Telescience Rework

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
tweak: Telescience now works with direct coordinates, parallel (x) and trasversal (y). To balance this, it needs to be calibrated to target a new location, which takes longer the further it is from the telepad or the last teleport point.
tweak: Telepads now also require a micro-manipulator, which will reduce the calibration time, but not the teleport time.
balance: Teleporting now has a very visible overlay on both sides before activating, and has a five-second delay.
/:cl:

## What does this fix tho
It minimizes the 'no counterplay' part of telesci: calibrating is visible to people near the target area (in the form of small pulses, followed by a larger pulse on the target point) giving them time to prepare to respond. Teleporting is even more visible and allows people to counteract it by moving items or preparing traps. For example, a hop who saw his locker getting stolen could hide inside, then lase the thieving scientist to death as soon as he comes close.

It also reduces the mobility of the telepad, especially on long distances. Teleporting someone into the supermatter is possible, but takes some time.

Kidnapping someone alive and moving was already pretty much a lost cause, but now the target pretty much has to be either willing or restrained.

And most importantly, it no longer requires an external tool to dictate the calibration time, which is now set in the code and can be balanced.

## Freeze!
I'd argue that this affects gameplay balance-wise less or the same as a complete removal. I don't really see why removals would be freeze-friendly at all, and i think they just got lumped in with reverts due to the tag Revert/Removal containing them both.

### CODE REVIEW

**Comment by duncathan** [comment]
"I don't see why removals are freeze friendly"
It's because they don't run the risk of adding bugs lmaoooooo

**Comment by duncathan** [comment]
also goof's idea was way better

**Comment by Iamgoofball** [comment]
>Telescience now works with direct coordinates, parallel (x) and trasversal (y). To balance this, it needs to be calibrated to target a new location, which takes longer the further it is from the telepad or the last teleport point.

it used to be this but offsets and it was so fucking easy to abuse

File: code/game/objects/effects/overlays.dm [hunk 1]
```diff
@@ -485,6 +485,31 @@
 	duration = 8
 	randomdir = 0
 
+/obj/effect/overlay/telesci
```
Comment (ChangelingRain) [inline code/game/objects/effects/overlays.dm]: why the fuck did you not use `/obj/effect/overlay/temp`, holy shit, you're in the same file as it

Comment (XDTM) [inline code/game/objects/effects/overlays.dm]: For some reason i thought that the duration had to be tied to the animation time, guess i'm just dumb

**Pull request closed without merging**